src/node.rs

```diff
@@ -1223,6 +1223,25 @@ impl<S: ForkSource + std::fmt::Debug> InMemoryNode<S> {
         }
     }
 
+    fn validate_tx(&self, tx: &L2Tx) -> Result<(), String> {
+        let max_gas = U256::from(u32::MAX);
+        if tx.common_data.fee.gas_limit > max_gas
```

Comment:
We should add unit tests at the bottom of this file for these additional checks. Example:

```rust
#[tokio::test]
async fn test_validate_tx_gas_limit_too_high() {
    // Arrange
    let node = InMemoryNode::<HttpForkSource>::default();
    let high_gas_limit = U256::from(u32::MAX) + U256::from(1);
    let tx_hash = H256::repeat_byte(0x01);
    let private_key = H256::random();
    let from_account = PackedEthSignature::address_from_private_key(&private_key)
        .expect("failed generating address");
    node.set_rich_account(from_account);

    // Act
    let mut tx = L2Tx::new_signed(
        Address::random(),
        vec![],
        Nonce(0),
        Fee {
            gas_limit: high_gas_limit,
            max_fee_per_gas: U256::from(250_000_000),
            max_priority_fee_per_gas: U256::from(250_000_000),
            gas_per_pubdata_limit: U256::from(20000),
        },
        U256::from(1),
        L2ChainId::from(260),
        &private_key,
        None,
        Default::default(),
    )
    .unwrap();
    tx.set_input(vec![], tx_hash);
    let result = node.apply_txs(vec![tx]);

    // Assert
    assert_eq!(result.err(), Some("exceeds block gas limit".into()));
}
```

Reply:
I was looking to add a test for the changes, but there are no tests for `send_raw_transaction`, I think I can add a test for the entire thing.